vars/piperStageWrapper.groovy

```diff
@@ -0,0 +1,107 @@
+
+    // load default & individual configuration
+    Map config = ConfigurationHelper.newInstance(this)
+        .loadStepDefaults()
+        .mixin(ConfigurationLoader.defaultStageConfiguration(this, stageName))
+        .mixinGeneralConfig(script.commonPipelineEnvironment)
+        .mixinStageConfig(script.commonPipelineEnvironment, stageName)
+        .mixin(parameters)
+        .use()
+
+    stageLocking(config) {
+        withNode(config) {
+            try {
+                utils.unstashAll(config.stashContent)
+
+                if (Boolean.valueOf(env.ON_K8S) && containerMap.size() > 0) {
+                    withEnv(["POD_NAME=${stageName}"]) {
+                        dockerExecute(script: script, containerMap: containerMap) {
+                            executeStage(body, stageName, config)
+                        }
+                    }
+                } else {
+                    executeStage(body, stageName, config)
+                }
+            } finally {
+                deleteDir()
+            }
+        }
+    }
+}
+
+String stageExitFilePath(String stageName, Map config) {
+    return "${config.extensionLocation}${stageName.replace(' ', '_').toLowerCase()}.groovy"
+}
+
+String globalStageExitFilePath(String stageName, Map config) {
+    return "${config.globalExtensionLocation}${stageName.replace(' ', '_').toLowerCase()}.groovy"
+}
+
+void stageLocking(Map config, Closure body) {
+    if (config.stageLocking) {
+        lock(resource: "${env.JOB_NAME}/${config.ordinal}", inversePrecedence: true) {
+            milestone config.ordinal
+            body()
+        }
+    } else {
+        body()
+    }
+}
+
+void withNode(Map config, Closure body) {
+    if (config.withNode) {
+        node(config.nodeLabel) {
+            body()
+        }
+    } else {
+        body()
+    }
+}
+
+void executeStage(originalStage, stageName, config) {
+
+    /* Defining the sources where to look for a project extension and a repository extension.
+    * Files need to be named like the executed stage to be recognized.
+    */
+    def projectInterceptorFile = stageExitFilePath(stageName, config)
+    def globalInterceptorFile = globalStageExitFilePath(stageName, config)
+    boolean projectExtensions = fileExists(projectInterceptorFile)
+    boolean globalExtensions = fileExists(globalInterceptorFile)
+    // Pre-defining the real originalStage in body variable, might be overwritten later if extensions exist
+    def body = originalStage
+
+    if (globalExtensions) {
+        Script globalInterceptorScript = load(globalInterceptorFile)
+        echo "[${STEP_NAME}] Found global interceptor '${globalInterceptorFile}' for ${stageName}."
+        // If we call the global interceptor, we will pass on originalStage as parameter
+        body = {
+            // passing config twice to keep compatibility with https://github.com/SAP/cloud-s4-sdk-pipeline-lib/blob/master/vars/runAsStage.groovy
```

Comment:
Maybe we can find better solutions than passing it twice? I might make sense ti discuss, whether users would want the configuration to be passed separately or not.

Reply:
I would be happy to pass it only once.
If there was a way to go this way, I would also prefer this.

Reply:
only passed once now ...